src/mito2/src/worker/handle_drop.rs

```diff
@@ -42,7 +42,28 @@ impl<S> RegionWorkerLoop<S> {
 
         info!("Try to drop region: {}", region_id);
 
-        // write dropping marker
+        // Marks the region as dropping.
+        region.set_dropping()?;
+
+        struct ResetState<'a> {
+            region: &'a MitoRegionRef,
+            discard: bool,
+        }
+
+        impl<'a> Drop for ResetState<'a> {
+            fn drop(&mut self) {
+                if !self.discard {
+                    self.region.switch_state_to_writable(RegionState::Dropping);
```

Comment:
Is it a good idea to set the state back to writable? The `discard` is set to "true" only after **all** changes to other things are succeeded. If something wrong happened in the middle, setting the state back to writable could put region in an impaired situation. For example, if `DROPPING_MARKER_FILE` is added, and the `region.stop()` is failed. Were the state of region set to "writable" again, it might cause some potential problem.

Reply:
`region.stop()` doesn't return an error so I changed its return value to `()`. Now only writing the marker may return an error. Setting the state back to writable so we can retry the operation.